runtime/browser/runtime_download_manager_delegate.cc

```diff
@@ -51,6 +55,14 @@ bool RuntimeDownloadManagerDelegate::DetermineDownloadTarget(
     content::DownloadItem* download,
     const content::DownloadTargetCallback& callback) {
   DCHECK(BrowserThread::CurrentlyOn(BrowserThread::UI));
+#if defined(OS_LINUX) && !defined(OS_TIZEN)
+  // Use the platform's default application to handle the download URL
+  if (default_download_path_.empty() &&
+      download->GetForcedFilePath().empty()) {
```

Comment:
The conditions are used to ensure the download is offloaded to platform's default URL download application when these is no download path preset. default_download_path_ is the path set by xwalk_browsertest only.